contracts/pricing/oracle/ChainlinkOracleManager.sol

```diff
@@ -117,31 +104,32 @@ contract ChainlinkOracleManager is
     {
         address assetOracle = getAssetOracle(_asset);
         IEACAggregatorProxy aggregator = IEACAggregatorProxy(assetOracle);
-        int256 answer = aggregator.latestAnswer();
+        (, int256 answer, , , ) = aggregator.latestRoundData();
         require(
             answer > 0,
             "ChainlinkOracleManager: No pricing data available"
         );
 
         return
-            uint256(answer)
-                .fromScaledUint(CHAINLINK_ORACLE_DECIMALS)
-                .toScaledUint(strikeAssetDecimals, true);
+            uint256(answer).fromScaledUint(aggregator.decimals()).toScaledUint(
+                strikeAssetDecimals,
+                true
+            );
     }
 
     /// @inheritdoc IProviderOracleManager
     function isValidOption(
-        address,
+        address _underlyingAsset,
         uint256,
         uint256
     )
-        public
+        external
         view
         virtual
         override(ProviderOracleManager, IProviderOracleManager)
         returns (bool)
     {
-        return true;
+        return assetOracles[_underlyingAsset] != address(0);
```

Comment:
Add call to `super.isValidOption()` in `ChainlinkFixedTimeOracle` which inherits from this